programs/vote/src/vote_state/mod.rs

```diff
@@ -613,11 +642,17 @@ pub fn process_new_vote_state(
                     return Err(VoteError::ConfirmationRollBack);
                 }
 
-                current_vote_state_index += 1;
-                new_vote_state_index += 1;
+                current_vote_state_index = current_vote_state_index
+                    .checked_add(1)
+                    .expect("`current_vote_state_index` is bounded by `MAX_LOCKOUT_HISTORY`");
```

Comment:
nit: duplicating error messages like this could bring pain should we hit one on a release build w/o debug symbols

Reply:
+1 on unique messages for debugging